### PR TITLE
[docs] update supported golang versions and docker tags

### PR DESCRIPTION
**Pull request by v1v**
### What

Use a table format for the docker tags.
Update docker tags with their relevant tags.
Update golang versions.
Add 1.17 version

### Issue

Closes https://github.com/elastic/golang-crossbuild/issues/121

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,27 +14,30 @@ The base image used is Debian 9 (stretch) unless otherwise specified.
...
+| linux/{mips,mipsle} | **See above** | **See above** | **See above** | **See above** | **See above** | **See above** | `1.16.7-mips32` | `1.17.1-mips32` |
+| linux/s390x | `1.10.8-s390x` | `1.11.13-s390x` | `1.12.12-s390` | `1.13.12-s390` | `1.14.15-s390` | `1.15.14-s390` | `1.16.7-s390` | `1.17.1-s390` |
+| linux/{amd64,386} and windows/{amd64,386} (Debian 7 (see **below**)) |`1.10.8-main-debian7` | `1.11.13-main-debian7` | `1.12.12-main-debian7` | `1.13.12-main-debian7` | `1.14.15-main-debian7` | `1.15.14-main-debian7` | `1.16.7-main-debian7` | `1.17.1-main-debian7` |
+| linux/{amd64,386} and windows/{amd64,386} (Debian 8 (see **below**)) | `1.10.8-main-debian8` | `1.11.13-main-debian8` | `1.12.12-main-debian8` | `1.13.12-debian8` | `1.14.15-main-debian8` | `1.15.14-main-debian8` | `1.16.7-main-debian8` | `1.17.1-main-debian8` |
+| linux/{amd64,386} and windows/{amd64,386} (Debian 9 (see **below**)) | NA | NA | NA | NA | NA | `1.15.14-main-debian9` | `1.16.7-main-debian9` | `1.17.1-main-debian9` |
+| linux/{amd64,386} and windows/{amd64,386} (Debian 10 (see **below**)) | NA | NA | NA | NA | NA | `1.15.14-main-debian10` | `1.16.7-main-debian10` | ` ` |
```
Comment (v1v) [inline README.md:34]: `1.17.1-main-debian10` is not available in the docker registry. We might need to investigate the reason.

**Review by mdelapenya — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (v1v) [reply README.md:34]: The last merge commit contains the 1.17 golang support and failed when pushing into the docker registry:
- https://beats-ci.elastic.co/blue/organizations/jenkins/Beats%2Fgolang-crossbuild-mbp/detail/master/40/pipeline/3606

Comment (v1v) [reply README.md:34]:
```suggestion
| linux/{amd64,386} and windows/{amd64,386} (Debian 10 (see **below**)) | NA | NA | NA | NA | NA | `1.15.14-main-debian10` | `1.16.7-main-debian10` | `1.17.1-main-debian10` |
```

**Comment by kuisathaverat** [comment]
with every new version we should create a new release and add the tags there it is more direct and clear, we did it 2 versions ago https://github.com/elastic/golang-crossbuild/releases

**Comment by v1v** [comment]
> with every new version we should create a new release 

I like that, so I'll raise an issue to track this automation then.

**Review by kuisathaverat — APPROVED** [review]

**Comment by v1v** [comment]
Unrelated failures in the CI

**Pull request merged**